site/src/components/pages/hub/BlockDataContainer.tsx

```diff
@@ -55,6 +71,46 @@ export const BlockDataContainer: VoidFunctionComponent<
     prevPackage.current = metadata.packagePath;
   }, [example, metadata.packagePath, text]);
 
+  useEffect(() => {
+    if (
+      metadata.variants &&
+      previousBlockVariantsTab.current !== blockVariantsTab
+    ) {
```

Comment:
If the user switches variants, delete old properties from the parsed text, append the new example and properties, and set it to the `text` state.